wls-auth-service/src/test/java/de/muenchen/oss/wahllokalsystem/authservice/service/CryptoServiceTest.java

```diff
@@ -0,0 +1,160 @@
+package de.muenchen.oss.wahllokalsystem.authservice.service;
+
+import de.muenchen.oss.wahllokalsystem.authservice.exception.ExceptionConstants;
+import de.muenchen.oss.wahllokalsystem.wls.common.exception.TechnischeWlsException;
+import de.muenchen.oss.wahllokalsystem.wls.common.exception.util.ServiceIDFormatter;
+import java.util.Base64;
+import java.util.Set;
+import java.util.stream.Stream;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import lombok.val;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CryptoServiceTest {
+
+    private static final String ENCRYPTION_PREFIX = "encryptionPrefix";
+
+    @Mock
+    ServiceIDFormatter idFormatter;
+
+    @Mock
+    Cipher cipher;
+
+    @InjectMocks
+    CryptoService unitUnderTest;
+
+    @BeforeEach
+    void setUp() {
+        unitUnderTest.setEncryptedPrefix(ENCRYPTION_PREFIX);
+    }
+
+    @Nested
+    class IsEncrypted {
+
+        @Test
+        void should_returnTrue_when_valueStartsWithPrefix() {
+            Assertions.assertThat(unitUnderTest.isEncrypted(ENCRYPTION_PREFIX + "the encrypted value")).isTrue();
+        }
+
+        @Test
+        void should_returnFalse_when_valueDoesNotStartWithPrefix() {
+            unitUnderTest.setEncryptedPrefix("prefix");
+            Assertions.assertThat(unitUnderTest.isEncrypted("values without encryption prefix")).isFalse();
+        }
```

Comment:
der `ENCRYPTION_PREFIX` wird in `beforeEach` ja für alle Tests gesetzt. Warum muss er dann in Zeile 50 zusätzlich manuell übergeben werden? Bzw warum  wird er in 55-56, nachdem er neu gesetzt wurde, automatisch erkannt und muss nicht mehr manuell übergeben werden?

Reply:
um sicherzugehen dass die Daten zur entschlüsslung nicht mit dem Prefix beginnen. Ich finde man erkennt auf diese Weise auch leichter dass die Daten nicht mit den Prefix beginnen weil man nicht zur Konstante springen muss.